libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1663,15 +1663,21 @@ void ExpressionCompiler::appendExternalFunctionCall(
 		utils().storeInMemoryDynamic(IntegerType(8 * CompilerUtils::dataStartOffset), false);
 	}
 	// If the function takes arbitrary parameters, copy dynamic length data in place.
+	solAssert(_functionType.takesArbitraryParameters() != _functionType.padArguments(), "");
```

Comment:
Actually this combination is used in `BareCall`: `<address>.call(1, 2, "test")`

Unfortunate that this simplification can't be used yet :)